File: basics/Q60/main.go
package main

import (
	"fmt"
	"reflect"
)

type User struct {
	ID        int
	FirstName string
	LastName  string
}

func main() {
	var user User = User{
		ID:        1,
		FirstName: "Elyar",
		LastName:  "Sadig",
	}
	var a1 int = 10
	var a2 int64 = 10
	var s string = "Hello World"
	sliceOfBytes := make([]byte, 1024)
	var r rune = 'A'
	var array [10]int

	fmt.Println("Memory Size of a1 is", getSize(a1), "bytes")
	fmt.Println("Memory Size of a2 is", getSize(a2), "bytes")
	fmt.Println("Memory Size of s is", getSize(s), "bytes")
	fmt.Println("Memory Size of user is", getSize(user), "bytes")
	fmt.Println("Memory Size of sliceOfBytes is", getSize(sliceOfBytes), "bytes")
	fmt.Println("Memory Size of r is", getSize(r), "bytes")
	fmt.Println("Memory Size of array is", getSize(array), "bytes")
}

func getSize(input any) uintptr {
	t := reflect.TypeOf(input)
	return t.Size()
}

File: basics/Q60/main.rs
use std::mem::size_of;

struct User {
    id: i32,
    first_name: String,
    last_name: String,
}

fn main() {
    let user = User {
        id: 1,
        first_name: "Elyar".to_string(),
        last_name: "Sadig".to_string(),
    };
    let a1: i32 = 10;
    let a2: i64 = 10;
    let s: String = "Hello World".to_string();
    let slice_of_bytes: Vec<u8> = vec![0u8; 1024];
    let r: char = 'A';
    let array: [i32; 10] = [0; 10];

    println!("Memory Size of a1 is {} bytes", size_of::<i32>());
    println!("Memory Size of a2 is {} bytes", size_of::<i64>());
    println!("Memory Size of s is {} bytes", size_of::<String>());
    println!("Memory Size of user is {} bytes", size_of::<User>());
    println!(
        "Memory Size of sliceOfBytes is {} bytes",
        size_of::<Vec<u8>>()
    );
    println!("Memory Size of r is {} bytes", size_of::<char>());
    println!("Memory Size of array is {} bytes", size_of::<[i32; 10]>());

    let _ = (user, a1, a2, s, slice_of_bytes, r, array);
}